### ISSUE TITLE
Strange error

### ISSUE THREAD

**Issue by sj-choi** [CompImg/LST-AI#10]
I encountered the following error after a successful LST-AI installation. 
Can anyone help me resolve this issue?

Installing collected packages: LST-AI
  Running setup.py develop for LST-AI
Successfully installed LST-AI-1.0.0
(py38) xubuntu@xubuntu-VirtualBox:~/LST-AI$ cd ..
(py38) xubuntu@xubuntu-VirtualBox:~$ lst -h
Illegal instruction (core dumped)

Thank you,
-SC

**Comment by jqmcginnis** [comment]
Hi @sj-choi 

Can you please share the entire output from the installation?

Can you please provide more information regarding your platform? Are you installing this in a virtual box?

**Comment by sj-choi** [comment]
Yes, I successfully installed LST-AI on one of my VirtualBox VMs, Xubuntu 18.04 LTS, on a Windows 11 host. I processed a few examples successfully and the results were great. Then, I decided to install LST-AI on another VirtualBox VM on a Windows 10 host. This time the installation appeared to be successful. However, when I tried to check if it worked well using the command 'lst -h' it issued the error message "Illegal instruction (core dumped)." This is an infrequent error message. Both Windows 11 and Windows 10 hosts are equipped with an identical Intel CPU. They are not very old and were purchased at a similar time. 
In summary, both Windows hosts were purchased at a similar time and are equipped with identical Intel CPUs (Intel(R) Xeon(R) W-2123 CPU @ 3.60GHz   3.60 GHz). One installation on Xubuntu VM on Windows 11 works and another installation on identical Xubuntu VM on Windows 10 host does not work. 
Is there any known installation trouble on VirtualBox VMs?

FYI, I share the whole installation output below:
(py38) xubuntu@xubuntu-VirtualBox:~$ git clone https://github.com/CompImg/LST-AI/
Cloning into 'LST-AI'...
remote: Enumerating objects: 380, done.
remote: Counting objects: 100% (33/33), done.
remote: Compressing objects: 100% (22/22), done.
remote: Total 380 (delta 14), reused 12 (delta 11), pack-reused 347
Receiving objects: 100% (380/380), 32.76 MiB | 16.26 MiB/s, done.
Resolving deltas: 100% (183/183), done.
(py38) xubuntu@xubuntu-VirtualBox:~$ cd LST-AI
(py38) xubuntu@xubuntu-VirtualBox:~/LST-AI$ pip install -e .
Obtaining file:///home/xubuntu/LST-AI
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy<1.24.4 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from LST-AI==1.0.0) (1.24.3)
Requirement already satisfied: pillow in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from LST-AI==1.0.0) (10.2.0)
Requirement already satisfied: scipy>=1.9.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from LST-AI==1.0.0) (1.10.1)
Requirement already satisfied: scikit-image>=0.21.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from LST-AI==1.0.0) (0.21.0)
Requirement already satisfied: tensorflow>=2.13 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from LST-AI==1.0.0) (2.13.1)
Requirement already satisfied: nibabel in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from LST-AI==1.0.0) (5.2.0)
Requirement already satisfied: requests in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from LST-AI==1.0.0) (2.26.0)
Requirement already satisfied: networkx>=2.8 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from scikit-image>=0.21.0->LST-AI==1.0.0) (3.1)
Requirement already satisfied: imageio>=2.27 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from scikit-image>=0.21.0->LST-AI==1.0.0) (2.34.0)
Requirement already satisfied: tifffile>=2022.8.12 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from scikit-image>=0.21.0->LST-AI==1.0.0) (2023.7.10)
Requirement already satisfied: PyWavelets>=1.1.1 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from scikit-image>=0.21.0->LST-AI==1.0.0) (1.4.1)
Requirement already satisfied: packaging>=21 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from scikit-image>=0.21.0->LST-AI==1.0.0) (21.0)
Requirement already satisfied: lazy_loader>=0.2 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from scikit-image>=0.21.0->LST-AI==1.0.0) (0.3)
Requirement already satisfied: absl-py>=1.0.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (2.1.0)
Requirement already satisfied: astunparse>=1.6.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (1.6.3)
Requirement already satisfied: flatbuffers>=23.1.21 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (23.5.26)
Requirement already satisfied: gast<=0.4.0,>=0.2.1 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (0.4.0)
Requirement already satisfied: google-pasta>=0.1.1 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (0.2.0)
Requirement already satisfied: grpcio<2.0,>=1.24.3 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (1.62.0)
Requirement already satisfied: h5py>=2.9.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (3.1.0)
Requirement already satisfied: keras<2.14,>=2.13.1 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (2.13.1)
Requirement already satisfied: libclang>=13.0.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (16.0.6)
Requirement already satisfied: opt-einsum>=2.3.2 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (3.3.0)
Requirement already satisfied: protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.20.3 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (4.25.3)
Requirement already satisfied: setuptools in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (50.3.1.post20201107)
Requirement already satisfied: six>=1.12.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (1.15.0)
Requirement already satisfied: tensorboard<2.14,>=2.13 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (2.13.0)
Requirement already satisfied: tensorflow-estimator<2.14,>=2.13.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (2.13.0)
Requirement already satisfied: termcolor>=1.1.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (1.1.0)
Requirement already satisfied: typing-extensions<4.6.0,>=3.6.6 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (4.5.0)
Requirement already satisfied: wrapt>=1.11.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (1.12.1)
Requirement already satisfied: tensorflow-io-gcs-filesystem>=0.23.1 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorflow>=2.13->LST-AI==1.0.0) (0.34.0)
Requirement already satisfied: importlib-resources>=1.3 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from nibabel->LST-AI==1.0.0) (6.1.2)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from requests->LST-AI==1.0.0) (1.26.6)
Requirement already satisfied: certifi>=2017.4.17 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from requests->LST-AI==1.0.0) (2021.10.8)
Requirement already satisfied: charset-normalizer~=2.0.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from requests->LST-AI==1.0.0) (2.0.3)
Requirement already satisfied: idna<4,>=2.5 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from requests->LST-AI==1.0.0) (3.2)
Requirement already satisfied: wheel<1.0,>=0.23.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from astunparse>=1.6.0->tensorflow>=2.13->LST-AI==1.0.0) (0.35.1)
Requirement already satisfied: zipp>=3.1.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from importlib-resources>=1.3->nibabel->LST-AI==1.0.0) (3.6.0)
Requirement already satisfied: pyparsing>=2.0.2 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from packaging>=21->scikit-image>=0.21.0->LST-AI==1.0.0) (3.0.4)
Requirement already satisfied: google-auth<3,>=1.6.3 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (2.28.1)
Requirement already satisfied: google-auth-oauthlib<1.1,>=0.5 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (1.0.0)
Requirement already satisfied: markdown>=2.6.8 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (3.3.4)
Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (0.7.2)
Requirement already satisfied: werkzeug>=1.0.1 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (2.0.1)
Requirement already satisfied: cachetools<6.0,>=2.0.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from google-auth<3,>=1.6.3->tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (4.2.2)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from google-auth<3,>=1.6.3->tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (0.2.8)
Requirement already satisfied: rsa<5,>=3.1.4 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from google-auth<3,>=1.6.3->tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (4.7.2)
Requirement already satisfied: requests-oauthlib>=0.7.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from google-auth-oauthlib<1.1,>=0.5->tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (1.3.0)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (0.4.8)
Requirement already satisfied: oauthlib>=3.0.0 in /home/xubuntu/anaconda3/envs/py38/lib/python3.8/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<1.1,>=0.5->tensorboard<2.14,>=2.13->tensorflow>=2.13->LST-AI==1.0.0) (3.1.1)
Installing collected packages: LST-AI
  Running setup.py develop for LST-AI
Successfully installed LST-AI-1.0.0
(py38) xubuntu@xubuntu-VirtualBox:~/LST-AI$ cd ..
(py38) xubuntu@xubuntu-VirtualBox:~$ lst -h
Illegal instruction (core dumped)
(py38) xubuntu@xubuntu-VirtualBox:~$

**Comment by sj-choi** [comment]
I've verified the installation of "HD-BET" and "greedy" using the provided commands as directed, and they appears to be installed without issues. However, I'm encountering an "Illegal Instruction" error message when running "lst."

**Comment by sj-choi** [comment]
I was unable to determine the solution to the problem I encountered with my VirtualBox virtual machine (VM). While it functions on some VMs, it fails on others. I created additional VMs using VMware, all of which operate smoothly. I'm uncertain as to why all VMware VMs function well, while only a handful of VirtualBox VMs do. It's likely more effective to run LST-AI on native Linux machines rather than on VMs. If, like me, you need to operate LST-AI on a VM, I would advise using VMware instead of VirtualBox.

**Comment by jqmcginnis** [comment]
@sj-choi Sorry I was so unrsponsive the last two weeks, we had some deadline to work on. Thank you for reporting back, I am sure this is going to be helpful for other users with VM issues :)
